java/code/src/com/redhat/rhn/domain/contentmgmt/ContentProjectFactory.java

```diff
@@ -337,6 +367,23 @@ public static void purgeTarget(EnvironmentTarget target) {
         return getSession().createQuery(query).uniqueResultOptional();
     }
 
+    /**
+     * Look up {@link SoftwareEnvironmentTarget}s in {@link ContentProject} whose {@link Channel}s have given suffix.
+     *
+     * @param suffix the channel suffix
+     * @param project the project
+     * @return the successor of the channel in the project
+     */
+    public static Set<SoftwareEnvironmentTarget> lookupSwTargetsWithSuffix(String suffix, ContentProject project) {
+        return new HashSet<>(HibernateFactory.getSession()
+                .createQuery("SELECT tgt FROM SoftwareEnvironmentTarget tgt " +
+                        "WHERE tgt.contentEnvironment.contentProject = :project " +
+                        "AND tgt.channel.label LIKE :suffix")
+                .setParameter("project", project)
+                .setParameter("suffix", "%" + suffix)
```

Comment:
Question: is it possible with some tricky naming of channels, that this query match wrong channels?

Reply:
This query only takes into account channels that are in given content project. But in general, the naming convention is something that can break the CLM functionality (we generate the channel labels based on `project-environment-originalLabel` convention and if somebody cleverly named their channels, so that they collide with this, problems will happen).

Reply:
We can think about adding a test in CLM when we add channels to a project and test if the name already contains "project-" or "-environment-" strings as part of the label. If this is the case, we should not allow to use this channel in this project.
But this is for another PR. Not this one.